protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,381 @@
+        uint256 index
+    ) public pure override returns (bytes32) {
+        return keccak256(abi.encodePacked(holder, index));
+    }
+
+    /**
+     * @notice Check if vesting Id exists
+     * @param _id Vesting Id
+     * @return Bool
+     */
+     function vestingIdExists(bytes32 _id) public view returns (bool) {
+        return _vestingIds.contains(_id);
+    }
+
+    /**
+     * @notice Check if vesting is revoked
+     * @param _vestingId Vesting Id
+     * @return Bool
+     */
+    function isVestingRevoked(bytes32 _vestingId) external view override returns (bool) {
+        if(!vestingIdExists(_vestingId)) { return true; }
+        return schedules[_vestingId].revoked;
+    }
+
+    /**
+     * @notice Get vesting schedule
+     * @param _vestingId Vesting Id
+     * @return schedule VestingSchedule 
+     */
+    function getSchedule(bytes32 _vestingId) external view override returns (VestingSchedule memory schedule) {
+        schedule = schedules[_vestingId];
+    }
+
+    /**
+     * @notice Returns the last vesting schedule for a given holder address.
+     * @param _recipient Recipient address
+     * @return VestingSchedule
+     */
+    function getLastVestingScheduleForHolder(
+        address _recipient
+    ) external view override returns (VestingSchedule memory) {
+        return
+            schedules[
+                computeVestingScheduleIdForAddressAndIndex(
+                    _recipient,
+                    _holdersVestingCount[_recipient] - 1
+                )
+            ];
+    }
+
+    /**
+     * @notice Returns vesting schedule by address and index
+     * @param recipient Recipient
+     * @param index Index
+     * @return VestingSchedule
+     */
+    function getVestingScheduleByAddressAndIndex(
+        address recipient,
+        uint256 index
+    ) external view override returns (VestingSchedule memory) {
+        return schedules[computeVestingScheduleIdForAddressAndIndex(recipient, index)];
+    }
+
+    /**
+     * @notice Get total vested amount at current block timestamp for vesting Id
+     * @param _vestingId Vesting Id
+     * @return Vested amount 
+     */
+    function getTotalVestingAtCurrentTime(bytes32 _vestingId) external view override returns (uint256) {
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        if (_schedule.amount == 0) { return 0; }
+        if (block.timestamp <= _schedule.start) { return 0; }
+        if (_schedule.revoked) { return 0; }
+        uint32 _elapsed = _getElapsedTime(_schedule.start, uint32(block.timestamp), _schedule.duration);
+        return TempleMath.mulDivRound(_schedule.amount, _elapsed, _schedule.duration, false);
+    }
+
+    /**
+     * @notice Get vesting summary
+     * @param _from From timestamp
+     * @param _to To timestamp
+     * @return summary Vesting summary 
+     */
+    function getVestingSummary(uint32 _from, uint32 _to) external view override returns (VestingSummary[] memory summary) {
```

Comment:
Would it make sense to have this function take a list of vestingId's it wants the details for?

Eg subgraph could index the `ScheduleCreated` events (and filter for the active dates), and then the client can choose if it passes in all those id's or splits it up across multiple RPC calls?

That way you can avoid any potential gas griefing issues and would make this function simpler?

Reply:
You're right, that's more efficient